Realm/RLMRealmConfiguration+Sync.mm

```diff
@@ -53,8 +53,11 @@ - (void)setSyncConfiguration:(RLMSyncConfiguration *)syncConfiguration {
 
     if (syncConfiguration.customFileURL) {
         self.config.path = syncConfiguration.customFileURL.path.UTF8String;
+    } else if (self.config.sync_config->flx_sync_requested) {
+        std::string str = std::string([@"default" UTF8String]);
```

Comment:
This is a really roundabout way of writing `"default"`.

Can a single app support both flexible sync and partition-based sync? If so using a legal partition name as the name of the flexible sync Realm seems like an issue.

Reply:
No, a single app cannot support both and I believe there is no intention for that to ever be the case.